### ISSUE TITLE
Sostituire l'ora 00:00 con le 24:00 del giorno prima

### ISSUE THREAD

**Issue by marcenuc** [iGruppi/iGruppi#103]
Quando si impostano gli orari di apertura e chiusura di un ordine, trovo molto poco intuitivo avere come intervallo di ore dalle `00:00` alle `23:59`, perché sono abituato a indicare la mezzanotte come le ore `24:00` e l'inizio del giorno come le ore `00:01`.

Anche una nostra referente, volendo chiudere un ordine a mezzanotte, ha messo come ora di chiusura le `00:00`, provocando la chiusura anticipata di 24 ore rispetto all'ora desiderata. Anche i gasisti hanno interpretato male l'ora di chiusura e si son trovati l'ordine chiuso, con conseguente scambio di mail.

Ovviamente, dopo aver spiegato l'errore, lo hanno capito. Ma trovo comunque più intuitivo avere come orari dalle `00:01` alle `24:00` e non dover spiegare come funziona a ogni nuovo gasista.

Che ne pensate? Andrebbe bene come modifica?

**Comment by Jazzo** [comment]
Per quanto mi riguarda credo sia un po' troppo soggettiva come richiesta.
Inoltre ormai lavoriamo così in ambiente di produzione da quasi un anno e
la modifica potrebbe creare confusione.
Secondo me dovrà rientrare anche questa tra quelle personalizzazioni che
ogni gruppo potrà gestire per proprio conto.

**Comment by StefanoMenozzi** [comment]
Ciao,
sono abbastanza certo, ma potrei approfondire il ragionamento che le 00:00 esistano mentre le 24:00 non esistano. Mi spiego meglio: salvo smentite, il giorno dura dalle 00.00 incluse alle 23.59 (24:00 escluse). Non compaiono mai in nessun orologio le 24.00. Penso che nel tempo il software andrà arricchito con commenti ed informazioni/note mentre si eseguono le varie operazioni ma da un punto di vista "scientifico" si debba essere il più precisi possibili.

**Comment by RobertoCardarelli** [comment]
Anche io direi che generalmente non esiste 24:00 e il giorno va dalle 00:00 alle 23:59. Dopo le 23:59:59 scatta l'ora 00:00 e poi si riparte: 00:00:01... Quindi anche io sarei per lasciare così com'è l'impostazione generale ed eventualmente lasciare libertà di personalizzazione ai vari gruppi.

**Comment by marcenuc** [comment]
Sì, ho proposto una "soluzione" invece di spiegare il problema... un classico errore da principiante nella segnalazione di bug, chiedo venia.

Il problema è:

1. chi apre l'ordine, e vuole chiuderlo alla mezzanotte del giorno `X`, deve impostare come giorno di chiusura `X+1`.
2. il gasista che guarda il giorno di chiusura dell'ordine, vede il giorno `X+1` e non si rende conto che la chiusura è il giorno `X`.

Eviterei di rendere la cosa configurabile, secondo me non ne vale la pena, anche perché può creare non poca confusione con gli ordini condivisi.

Quanto alla proposta di @StefanoMenozzi di aggiungere note e segnalazioni, non saprei bene cosa segnalare. _"Attenzione, la chiusura è per il giorno `X-1`, NON per il giorno `X`"_? Tra l'altro, noto ora che, nell'elenco degli ordini, l'ora non è proprio indicata. Quindi leggo _"Ordine Chiuso il 20/04/2016"_ quando in realtà ha chiuso il 19 aprile.

Forse la soluzione è semplicemente di mettere in grassetto l'ora, magari più grande del giorno, e visualizzarla sempre. Che ne pensate?

**Comment by Jazzo** [comment]
@marcenuc credo tu abbia approfondito la questione ma il nocciolo rimane lo stesso: segnalare ai propri utenti che esistono degli standard internazionali a cui fare riferimento. Non ce li siamo inventati noi.
Certo, aggiungere l'ora potrebbe aiutare ma non credo sia la soluzione. La soluzione è far capire alle persone come funziona lo standard di riferimento.

**Comment by marcenuc** [comment]
Non sono stato chiaro: ho ritirato la mia proposta iniziale! è stata una proposta buttata lì senza un'adeguata riflessione. Quindi non ho alcuna intenzione di mettere mano a nessuno standard internazionale. Ho cambiato il titolo nel tentativo chiarire questo aspetto.

>> Quindi leggo "Ordine Chiuso il 20/04/2016" quando in realtà ha chiuso il 19 aprile.
> 
> No, ha chiuso il 20 Aprile alle ore 00:00.

Scusa ma questo non l'ho capito: da cosa lo capisci che è chiuso alle ore 00:00? Stai dicendo che lo standard è che, se l'ora non è indicata allora è 00:00?

**Comment by Jazzo** [comment]
>
> Scusa ma questo non l'ho capito: da cosa lo capisci che è chiuso alle ore
> 00:00? Stai dicendo che lo standard è che, se l'ora non è indicata allora è
> 00:00?
>

No, non so quale sia l'orario.  Dico solo che se appare la data del
20/04/16 l'orario di chiusura sarà dalle 00:00 alle 23:59. In tutti questi
casi non sarà mai il 19 Aprile quindi quello che hai scritto (vedi sopra) è
sbagliato.

Detto ciò se la soluzione è visualizzare l'ora non vedo perché continuare e
discutere anziché scrivere il codice.
